File: sources/be-springboot/favUrls/src/main/java/cz/osu/vbap/favUrls/model/entities/AppUser.java
package cz.osu.vbap.favUrls.model.entities;

import jakarta.persistence.*;
import lombok.*;
import org.jetbrains.annotations.Contract;
import cz.osu.vbap.favUrls.lib.ArgVal;

import java.util.Collection;

@Getter
@Setter
@NoArgsConstructor
@Entity
public class AppUser {
  @Id
  @GeneratedValue(strategy = jakarta.persistence.GenerationType.IDENTITY)
  private int appUserId;
  @Column(unique = true, nullable = false, length = 64)
  private String email;
  private String passwordHash;

  @OneToMany(mappedBy = "appUser", fetch = FetchType.LAZY)
  private Collection<Url> urls;

  @OneToMany(mappedBy = "appUser", fetch = FetchType.LAZY)
  private Collection<Tag> tags;

  @OneToMany(mappedBy = "appUser", fetch = FetchType.LAZY)
  private Collection<Token> tokens;

  @Contract(pure = true)
  public AppUser(@NonNull String email) {
    ArgVal.matchRegex(email, ".+@.+", "email");
    this.email = email;
  }

  @PrePersist
  private void prePersistCheck(){
    if (email != null)
      email = email.toLowerCase();
  }
}

File: sources/be-net/FavUrls/Model/Entities/AppUser.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace FavUrls.Model.Entities
{
  public class AppUser
  {
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int AppUserId { get; set; }
    public string Email { get; set; } = null!;
    public string PasswordHash { get; set; } = null!;
    public List<Token> Tokens { get; set; } = [];
    public List<Url> Urls { get; set; } = [];
    public List<Tag> Tags { get; set; } = [];
  }
}
